### ISSUE TITLE
Filtering relations with invalid relation_state returns an empty list instead of error message

### ISSUE THREAD

**Issue by isabelcosta** [anitab-org/mentorship-backend#449]
**Describe the bug**

When `relation_state` is a number (invalid filter) the API returns an empty list instead of a dictionary with an error message.

**To Reproduce**

Pre-requisite: have a user registered

1. Login the user
2. Access `GET /mentorship_relations` and filter by `relation_state`, with the value 2
3. See the empty list as the response and 400 HTTP error code.

**Expected behavior**

If the HTTP code is 400, for BAD REQUEST, I think the response should contain a message explaining that the `relation_state` filter is not valid.

**Screenshots**

Screenshot with `400 BAD REQUEST` and an empty list being returned when `relation_state` have value `2`.

<img width="1071" alt="Screenshot 2020-02-15 at 19 07 31" src="https://user-images.githubusercontent.com/11148726/74684662-5cc3d800-51c4-11ea-9005-6783148e0c87.png">

**Desktop (please complete the following information):**
 - OS: Mac OS
 - Browser: Chrome
 - Version 🤷‍♀ 

**Additional context**

🤷‍♀

**Comment by isabelcosta** [comment]
Is this a bug? Does this make sense? Should it continue to be an empty list?!
Asking for opinion @annabauza @BadduCoder

**Comment by sumit-badsara** [comment]
Yeah @isabelcosta , It would be better if an error is returned. We don't want it to be considered as "There are no requests currently". Because in that case too, list would be empty.

**Comment by PratibhaShrivastav** [comment]
@isabelcosta I would like to work on this!

**Comment by srivastava9** [comment]
@isabelcosta It Shows internal server error while registering for a new user,and while executing again it show error 400 username already exists.
I think it is not able to send email.
![Screenshot from 2020-02-21 14-47-49](https://user-images.githubusercontent.com/45288400/75020961-a90b6280-54b9-11ea-8996-9994044b365b.png)

**Comment by isabelcosta** [comment]
@PratibhaShrivastav sure you can work on this :)

**Comment by isabelcosta** [comment]
@srivastava9 you are correct. Sometimes we have this issue, and it is related to Gmail security settings, that can be wrongly set. This problem is explained in issue #233. I think I fixed the issue in the email we used to send emails, and once the server comes back up I think this will be solved. Thank you for reporting this bug @srivastava9 :)

**Comment by SanujBansal** [comment]
@isabelcosta I tried returning a dictionary with error message stated but while debugging i found that due to us using mashal_list_with functionality from flask for returning json array of mentorship relations we cannot do much about it without replacing the logic for returning relations.
![Screenshot from 2020-02-27 01-42-17](https://user-images.githubusercontent.com/32351832/75383513-779ff600-5902-11ea-85c2-d02bd65f0a18.png)

![Screenshot from 2020-02-27 01-27-13](https://user-images.githubusercontent.com/32351832/75382421-7a015080-5900-11ea-90df-c55b891af51e.png)

**Comment by satya7289** [comment]
@isabelcosta I want to work on this issue.

**Comment by mtreacy002** [comment]
@isabelcosta, I don't know if this can be considered as a bug. Because perhaps relation_id = 2 (Accepted) is considered not a request anymore? I tried to run this on the Android UI and it doesn't seem to be anything wrong.
So, this is what I've done.
- user testtwo sent request to testone and got accepted, they are now in relationship
- as user testtwo, if I look at my Requests tab, I have options: Pending, Past and All (I'm pretty sure 'All' means both Pending and Past, so not including current). At the time I did the test, I have no pending or past request. 

<img width="387" alt="Screen Shot 2020-04-03 at 3 37 52 pm" src="https://user-images.githubusercontent.com/29667122/78324691-1d5d1980-75c1-11ea-88a0-471be2157c07.png">

- But since I have 'currently' one mentorship relation, I can see it inside my 'Relation' tab (this shows 'Current Relation'.

<img width="394" alt="Screen Shot 2020-04-03 at 3 32 38 pm" src="https://user-images.githubusercontent.com/29667122/78324454-79736e00-75c0-11ea-9195-383312d66bd3.png">

ATM, we're only allowed to have one mentorship relation at a time, I think this is an expected behaviour? But even if we have more than one relation at a time, shouldn't it still best not to display accepted requests in the 'Requests' tab but in the 'Relation' tab?

This is just my thoughts.... 😉 . What do you all think?

**Comment by mtreacy002** [comment]
But then I noticed this, for a user who has no current relation and tried to send requests to more than one person (to widen their chance perhaps... 😀 ), only the last request got recorded and the previous one didn't (in my test I sent 2 requests).  I think we can consider this as a bug then, coz the expected behaviour is for the all requests to be shown in 'Pending' and 'All' list.

![Screen Shot 2020-04-03 at 4 12 54 pm](https://user-images.githubusercontent.com/29667122/78326404-08cf5000-75c6-11ea-9ece-20fdccb37ac9.png)

<img width="395" alt="Screen Shot 2020-04-03 at 4 12 36 pm" src="https://user-images.githubusercontent.com/29667122/78326412-108ef480-75c6-11ea-9040-e45684075c0e.png">

<img width="388" alt="Screen Shot 2020-04-03 at 4 26 32 pm" src="https://user-images.githubusercontent.com/29667122/78327193-079f2280-75c8-11ea-8d42-b98c8f6c4eb4.png">

**Comment by isabelcosta** [comment]
> But then I noticed this, for a user who has no current relation and tried to send requests to more than one person (to widen their chance perhaps... 😀 ), only the last request got recorded and the previous one didn't (in my test I sent 2 requests).  I think we can consider this as a bug then, coz the expected behaviour is for the all requests to be shown in 'Pending' and 'All' list.
(...)

Can you clarify what happened? there are 2 requests in the pending state but only one is shown both in the Pending and All tabs? @mtreacy002

**Comment by isabelcosta** [comment]
So right now, a user can only be in only mentorship relation at a time (in the ACCEPTED state). And if I am not mistaken from the backend API, all requests should be fetched including the Accepted one, but in the Android UI we may filter the current one, since its shown in a specific screen.

@mtreacy002 from the message above the one I quoted, what are yous suggestions?

**Comment by hongsungin92** [comment]
hello @isabelcosta ! have you not solved this issue yet?

**Comment by hongsungin92** [comment]
I think I fixed the code a little bit and solved this problem, can you check it? @isabelcosta

**Comment by sihhub** [comment]
@isabelcosta   I would like to work on this!

I have solved this problem now. So I can send the PR right away!

**Comment by isabelcosta** [comment]
@hongsungin92 this is not assigned to you yet, and I see I lost some messages from previous contributors who are interested, let me ask them first if they still wish to work on this.

**Comment by isabelcosta** [comment]
@PratibhaShrivastav are you still interested in working on this issue? I forgot to assign this to you at the time. let me know if this is not the case and I can assign it to someone else.

@srivastava9 @SanujBansal are you still interested?

**Comment by sihhub** [comment]
oh I see..! please check and let me know :)

**Comment by chahat99** [comment]
@isabelcosta can I work on this issue?

**Comment by isabelcosta** [comment]
@hongsungin92 are you still interested? Since I got no answers so far, I can totally assign this to you :)

@chahat99 let me check first with @hongsungin92

**Comment by sihhub** [comment]
@isabelcosta yes!  I am still interested in this issue:)
